test/coverlet.core.tests/coverlet.core.tests.csproj

```diff
@@ -30,7 +30,10 @@
     <ProjectReference Include="$(RepoRoot)test\coverlet.core.tests.samples.netstandard\coverlet.core.tests.samples.netstandard.csproj" />
     <ProjectReference Include="$(RepoRoot)test\coverlet.tests.xunit.extensions\coverlet.tests.xunit.extensions.csproj" />
     <ProjectReference Include="$(RepoRoot)test\coverlet.tests.projectsample.netframework\coverlet.tests.projectsample.netframework.csproj" />
-    <ProjectReference Include="$(RepoRoot)test\coverlet.tests.projectsample.vbmynamespace\coverlet.tests.projectsample.vbmynamespace.vbproj" />
+  </ItemGroup>
+  
+  <ItemGroup Condition="$([MSBuild]::IsOSPlatform('windows'))">
+      <ProjectReference Include="$(RepoRoot)test\coverlet.tests.projectsample.vbmynamespace\coverlet.tests.projectsample.vbmynamespace.vbproj" />
```

Comment:
Why did you add this condition here? It is possible to compile the assembly on linux.
![grafik](https://github.com/coverlet-coverage/coverlet/assets/19378678/ad7cf465-a45a-47cb-9953-1c53d0306719)

Reply:
coverlet.tests.projectsample.vbmynamespace.dll targets net48 platform which is only available for windows.  Therefore compilation is not necessary for Linux and MacOS. 

```Text
Testhost process for source(s) '/home/vsts/work/1/s/test/coverlet.tests.projectsample.vbmynamespace/bin/Debug/net48/coverlet.tests.projectsample.vbmynamespace.dll' exited with error: Unhandled Exception:
System.TypeInitializationException: The type initializer for 'Microsoft.VisualStudio.TestPlatform.ObjectModel.EqtTrace' threw an exception. ---> System.TypeLoadException: Could not load type of field 'Microsoft.VisualStudio.TestPlatform.ObjectModel.PlatformEqtTrace:TraceSourceLevelsMap' (1) due to: Could not resolve type with token 01000019 from typeref (expected class 'System.Diagnostics.TraceLevel' in assembly 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089') assembly:System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 type:System.Diagnostics.TraceLevel member:(null)
   --- End of inner exception stack trace ---
  at Microsoft.VisualStudio.TestPlatform.TestHost.Program.Main (System.String[] args) [0x00033] in <985811b3ce274022a2072f51d4b88753>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'Microsoft.VisualStudio.TestPlatform.ObjectModel.EqtTrace' threw an exception. ---> System.TypeLoadException: Could not load type of field 'Microsoft.VisualStudio.TestPlatform.ObjectModel.PlatformEqtTrace:TraceSourceLevelsMap' (1) due to: Could not resolve type with token 01000019 from typeref (expected class 'System.Diagnostics.TraceLevel' in assembly 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089') assembly:System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 type:System.Diagnostics.TraceLevel member:(null)
   --- End of inner exception stack trace ---
  at Microsoft.VisualStudio.TestPlatform.TestHost.Program.Main (System.String[] args) [0x00033] in <985811b3ce274022a2072f51d4b88753>:0 

```

Reply:
OK so `net48` can only be run on windows, but you can build a `net48` assembly on linux. The test `Instrumenter_MethodsWithoutReferenceToSource_AreSkipped ` also doesn't run this assembly, it only instruments it. And thus the test also executes on linux.

Sure it currently isn't a real scenario that `coverlet` instruments on a different operating system than the test assemblies are being executed. But technically it would be possible.

Nevertheless this wasn't really the objective here. In the end we wanted to test a scenario that an assembly/symbole contains a method definition that has no reference to a source (https://github.com/coverlet-coverage/coverlet/blob/master/src/coverlet.core/Instrumentation/Instrumenter.cs#L533-L535). This is a rather odd scenario but this `visual basic` `net48` assembly contains exactly such a scenario. So the test wasn't only meant for a `net48` assembly executed on windows.

This often is the way we analyze/fix bugs. Create an assembly that has the **problematic** construct inside, make coverlet that resilient that it can handle the problem and in the end write a unit test that checks the correct instrumentation. We actually do this because it is a pretty fast process. But I also agree that this process of writing tests could be optimized. E.g. by writing more actual unit tests instead of integration tests.

As this test currently doesn't make any problems on the other operating systems, can we agree to just leave it as it is?

Reply:
I removed the condition but the rational should be documented somewhere.